### PR TITLE
1.x: Publish code coverage results to codecov.io

### PR DESCRIPTION
**Pull request by zsxwing**
This is an example built in my repo: https://codecov.io/gh/zsxwing/RxJava/

### CODE REVIEW

**Comment by artem-zinnatullin** [comment]
Yes! I actually [started the same in my fork](https://github.com/artem-zinnatullin/RxJava/pull/1) previous week 😸.

@akarnokd: 2 most important things that codecov gives:
1. Code coverage for each PR and configurable commit status check, it'll report that some changes in files from PR are not covered and how overall coverage of the project was affected by PR.
2. At any time you can check coverage of particular file, take a look https://codecov.io/gh/zsxwing/RxJava/tree/1.x/src/main/java/rx .

We use it in several projects: [StorIO](https://codecov.io/github/pushtorefresh/storio?branch=master), [qualitymatters](https://codecov.io/github/artem-zinnatullin/qualitymatters?branch=master) and others, it makes testing more fun and improves quality of the project.

File: build.gradle [hunk 1]
```diff
@@ -6,6 +6,7 @@ buildscript {
 description = 'RxJava: Reactive Extensions for the JVM – a library for composing asynchronous and event-based programs using observable sequences for the Java VM.'
 
 apply plugin: 'java'
+apply plugin: 'jacoco'
```
Comment (artem-zinnatullin) [inline build.gradle:9]: Please specify version of jacoco, like this:

``` groovy
jacoco {
  toolVersion = '0.7.7.201606060606' // See http://www.eclemma.org/jacoco/.
}
```

Comment (zsxwing) [inline build.gradle:9]: Added.

**Comment by akarnokd** [comment]
:+1:

**Comment by artem-zinnatullin** [comment]
👍 yay, so happy about it, thx @zsxwing!

**Comment by zsxwing** [comment]
PR for 2.x is on the way :)

**Pull request merged**